src/main/java/org/csanchez/jenkins/plugins/kubernetes/volumes/ConfigMapVolume.java

```diff
@@ -24,26 +24,31 @@
 
 package org.csanchez.jenkins.plugins.kubernetes.volumes;
 
-import org.jenkinsci.Symbol;
-import org.kohsuke.stapler.DataBoundConstructor;
-
 import hudson.Extension;
 import hudson.model.Descriptor;
 import io.fabric8.kubernetes.api.model.Volume;
 import io.fabric8.kubernetes.api.model.VolumeBuilder;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
 
 public class ConfigMapVolume extends PodVolume {
 
     private String mountPath;
+    private String subPath;
     private String configMapName;
     private Boolean optional;
 
-    @DataBoundConstructor
-    public ConfigMapVolume(String mountPath, String configMapName, Boolean optional) {
+     @DataBoundConstructor
+    public ConfigMapVolume(String mountPath, String configMapName, Boolean optional, String subPath) {
```

Comment:
Instead of this, you could add a setter for `subPath` annotated with `@DataBoundSetter`

Reply:
Hello, viatombe,
I'm going to need a bit more explanation on why this is not the correct way of doing things. or how you preffer me to do it
I do not know why is wrong the go from 3 parameters in the constructor to 4.
I do not know the difference between @DataBoundConstructor and @DataboundSetter. 
however @DataBoundSetter was not used anywhere in the previous code.
and I'm just adding a new parameter to the constructor. keeping the old one for compatibility issues. 
I do not know where is the issue. please explain.
I'm very happy to learn the right way of doing things.

Reply:
You can read the explanation [here](https://www.jenkins.io/doc/developer/plugin-development/pipeline-integration/) -- Constructor vs. setters

Reply:
Thanks Vincent,
the link was quite useful and I understand the difference.
I've done the modifications on my original code/[PR](https://github.com/jenkinsci/kubernetes-plugin/pull/1024)
All the other issues are not in my original code/PR.
Please consider merging [PR](https://github.com/jenkinsci/kubernetes-plugin/pull/1024) while @hameno fixes the issues on his side.